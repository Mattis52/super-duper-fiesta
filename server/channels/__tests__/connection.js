jest.mock('child_process');
jest.mock('../../utils');
jest.mock('../../models/meeting');
jest.mock('../../models/issue');
jest.mock('../../models/vote');
jest.mock('../../models/user');
const { broadcast, emit } = require('../../utils');
const { execSync } = require('child_process');

execSync.mockImplementation(() => Buffer.from('fake_git_hash'));
const connection = require('../connection');
const { getActiveGenfors } = require('../../models/meeting');
const { getAnonymousUser } = require('../../models/user');
const { getVotes } = require('../../models/vote');
const { getActiveQuestion, getQuestions } = require('../../models/issue');
const { generateSocket, generateGenfors, generateAnonymousUser, generateIssue, generateVote } = require('../../utils/generateTestData');


describe('connection', () => {
  beforeEach(() => {
    getActiveGenfors.mockImplementation(async () => generateGenfors());
    getAnonymousUser.mockImplementation(
      async (passwordHash, onlinewebId, genfors) => generateAnonymousUser({
        passwordHash,
        genfors,
      },
    ));
    getActiveQuestion.mockImplementation(async () => generateIssue());
    getQuestions.mockImplementation(async meeting => [
      generateIssue({ meeting: meeting.id, _id: 2 }),
      generateIssue({ meeting: meeting.id, _id: 2 }),
      generateIssue({ meeting: meeting.id, _id: 2 }),
      generateIssue({ meeting: meeting.id, _id: 2 }),
    ]);
    getVotes.mockImplementation(async ({ _id: issueId }) => [
      generateVote({ question: issueId, _id: 1 }),
      generateVote({ question: issueId, _id: 2 }),
      generateVote({ question: issueId, _id: 3 }),
      generateVote({ question: issueId, _id: 4 }),
    ]);
  });

  it('emits correct actions when signed in and active genfors', async () => {
    await connection(generateSocket());

    expect(emit.mock.calls).toMatchSnapshot();
    expect(broadcast.mock.calls).toEqual([]);
  });
});
