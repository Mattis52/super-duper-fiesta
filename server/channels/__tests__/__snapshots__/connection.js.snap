// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connection emits correct actions when signed in and active genfors 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "VERSION",
    Object {
      "version": "fake_git_hash",
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "AUTH_SIGNED_IN",
    Object {
      "full_name": "Namy",
      "id": "123",
      "logged_in": true,
      "permissions": 5,
      "username": "123",
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "AUTH_REGISTERED",
    Object {
      "registered": true,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "OPEN_MEETING",
    Object {
      "id": 1,
      "pin": 3141592653,
      "registrationOpen": true,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "OPEN_ISSUE",
    Object {
      "_id": 1,
      "active": true,
      "description": "Description goes here",
      "genfors": "1",
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ENABLE_VOTING",
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 1,
      "option": 3,
      "question": 1,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 2,
      "option": 3,
      "question": 1,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 3,
      "option": 3,
      "question": 1,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 4,
      "option": 3,
      "question": 1,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "CLOSE_ISSUE",
    Object {
      "_id": 2,
      "active": true,
      "description": "Description goes here",
      "genfors": "1",
      "meeting": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "CLOSE_ISSUE",
    Object {
      "_id": 2,
      "active": true,
      "description": "Description goes here",
      "genfors": "1",
      "meeting": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "CLOSE_ISSUE",
    Object {
      "_id": 2,
      "active": true,
      "description": "Description goes here",
      "genfors": "1",
      "meeting": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "CLOSE_ISSUE",
    Object {
      "_id": 2,
      "active": true,
      "description": "Description goes here",
      "genfors": "1",
      "meeting": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 1,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 2,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 3,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 4,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 1,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 2,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 3,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 4,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 1,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 2,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 3,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 4,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 1,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 2,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 3,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
  Array [
    Object {
      "request": Object {
        "headers": Object {
          "cookie": Object {
            "passwordHash": "hashy",
          },
        },
        "user": Object {
          "_id": "123",
          "completedRegistration": true,
          "genfors": "1",
          "logged_in": true,
          "name": "Namy",
          "onlinewebId": "123",
          "permissions": 5,
        },
      },
    },
    "ADD_VOTE",
    Object {
      "_id": 4,
      "option": 3,
      "question": 2,
      "user": 1,
    },
  ],
]
`;
